backend/src/todo-service.js

```diff
@@ -56,3 +71,32 @@ export const deleteTodoItem = async (listId, itemId) => {
   await todoRepository.delete(itemId)
   return todoItem
 }
+
+export const reorderTodoItems = async (listId, itemIds) => {
+  const AppDataSource = await getAppDataSource()
+  const todoItemRepository = AppDataSource.getRepository(TodoItem)
+  
+  try {
+    const todoItems = await todoItemRepository.find({
+      where: { list: { id: listId } }
+    })
+    
+    // Use Promise.all for concurrent updates
+    await Promise.all(itemIds.map(async (itemId, index) => {
+      const todoItem = todoItems.find(item => item.id === parseInt(itemId))
+      
+      if (todoItem) {
+        todoItem.order = index
+        await todoItemRepository.save(todoItem)
+      }
+    }))
```

Comment:
```suggestion
    await Promise.all(itemIds.map((itemId, index) => {
      const todoItem = todoItems.find(item => item.id === parseInt(itemId))
      
      if (todoItem) {
        todoItem.order = index
        return todoItemRepository.save(todoItem)
      }
    }))
```

Awaiting inside the callback seem to defeat the purpose of `Promise.all()` here.

Reply:
You're right. Suggested change has been applied.